src/command/defaults/ClearCommand.php

```diff
@@ -0,0 +1,178 @@
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.permission"));
+				return true;
+			}
+
+			$target = $sender->getServer()->getPlayer($args[0]);
+		}
+
+		if($target === null){
+			$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.generic.player.notFound"));
+			return true;
+		}
+
+		$item = null;
+		if(isset($args[1])){
+			try{
+				$item = LegacyStringToItemParser::getInstance()->parse($args[1]);
+			}catch(InvalidArgumentException $e){
+				//vanilla checks this at argument parsing layer, can't come up with a better alternative
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.give.item.notFound", [$args[1]]));
+				return true;
+			}
+		}
+
+		$maxCount = -1;
+		if(isset($args[2])){
+			if(!is_numeric($args[2])){
+				$maxCount = -1;
+			}
+
+			if($item !== null){
+				$item->setCount($maxCount = $this->getInteger($sender, $args[2], 0));
+			}
+		}
+
+		//checking players inventory for all the items matching the criteria
+		if($item !== null and $item->getMeta() === 0 and $maxCount === 0){
+			$count = 0;
+			$contents = array_merge($target->getInventory()->all($item), $target->getArmorInventory()->all($item));
+			foreach($contents as $content){
+				$count += $content->getCount();
+			}
+
+			if($count > 0){
+				$sender->sendMessage(new TranslationContainer("%commands.clear.testing", [$target->getName(), $count]));
+			}else{
+				$sender->sendMessage(new TranslationContainer(TextFormat::RED . "%commands.clear.failure.no.items", [$target->getName()]));
+			}
+
+			return true;
+		}
+
+		$cleared = 0;
+
+		//clear everything from the targets inventory
+		if($item === null){
+			$contents = array_merge($target->getInventory()->getContents(), $target->getArmorInventory()->getContents());
+			foreach($contents as $content){
+				$cleared += $content->getCount();
+			}
+
+			$target->getInventory()->clearAll();
+			$target->getArmorInventory()->clearAll();
+			$target->getCursorInventory()->clearAll();
+		}else{
+			//clear the item from targets inventory irrelevant of the count
+			if($maxCount === -1){
+				foreach($target->getInventory()->all($item) as $index => $i){
+					$cleared += $i->getCount();
+					$target->getInventory()->clear($index);
+				}
+			}else{
+				//clear only the given amount of that particular item from players inventory
+				foreach($target->getInventory()->all($item) as $index => $i){
+					if($i->getCount() >= $maxCount){
+						$i->pop($maxCount);
+						$cleared += $maxCount;
+						$target->getInventory()->setItem($index, $i);
+						break;
+					}
+
+					if($maxCount <= 0){
+						break;
+					}
+
+					$cleared += $i->getCount();
+					$maxCount -= $i->getCount();
+					$target->getInventory()->clear($index);
+				}
+			}
+
+			//vanilla behaviour, removes item from armor inventory, regardless of the count specified
+			if(($index = $target->getArmorInventory()->first($item)) !== -1){
```

Comment:
I don't like this. This behaviour looks buggy and it doesn't make sense to replicate vanilla bugs.

Reply:
So what do you suggest as a replacement..? Just check if the item is

Reply:
I think a more sensible alternative would be to check all the available inventories for the item in a specific order. Maybe the armour inventory could be prioritized, but it doesn't make sense to remove items from multiple inventories when only a count of one was desired.

Reply:
Alright, seems fair :+1:

Reply:
Fixed, please review.